nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/PackageReferenceUpdater.cs

```diff
@@ -36,6 +36,28 @@ public static async Task UpdateDependencyAsync(
         // Get the set of all top-level dependencies in the current project
         var topLevelDependencies = MSBuildHelper.GetTopLevelPackageDependencyInfos(buildFiles).ToArray();
 
+        // if the name contains a semicolon, split it into two identical dependencies except with different names
+        foreach (var dependency in topLevelDependencies)
+        {
+            if (dependency.Name.Contains(';'))
```

Comment:
Can the split be moved to MSBuildHelper.GetTopLevelPackageDependencyInfos()?  It feels cleaner to only report simple/single dependencies from there.